darwinpush/client.py

```diff
@@ -143,15 +193,21 @@ def _connect(self):
         # while self.connected:
         #     time.sleep(1)
 
-    def _on_message(self, headers, message):
+    def on_ftp_message(self, message, source="FTP"):
+        self._on_message(None, message, source)
+
+    def _on_message(self, headers, message, source=None):
+
+        if type(message) == bytes:
+            message = message.decode("utf-8")
 
         # Decode the message and parse it as an XML DOM.
-        doc = domutils.StringToDOM(message.decode("utf-8"))
+        doc = domutils.StringToDOM(message)
```

Comment:
Is this needed? I pass a string to the CreateFromDom directly and works fine. Is it more safe to use this way?